sdk-setup/src/sdk-make-qmltypes

```diff
@@ -0,0 +1,1554 @@
+
+CONFIGURATION FILE
+    Each package is described on a separate line with fields separated by tabs
+    or spaces. Blanks and anything following '#' up to the end of line is
+    ignored.
+
+    The fields are
+
+        Source Package Name
+            This must match the "Name:" tag inside package's spec file. When
+            multiple spec files exist in a package, the one with matching
+            "Name:" tag is selected.
+
+        Normalized Git URL
+            This must be in the form
+                SERVICE ':' PATH
+            E.g. "git.merproject.org:mer-core/ssu". The PATH must be relative.
+
+        Git Branch
+            The Git branch to check out
+
+        Binary Packages
+            A comma separated list of the binary packages which provide the QML
+            modules to dump type information for. Each package can be listed
+            ether with (a) full name, (b) name starting with "-" which will be
+            treated as a suffix to the Package Name or (c) just "-" which
+            stands for name equal to the Package Name.
+
+    Example configuration file
+
+        # PACKAGE  REPOSITORY                       BRANCH  BINARIES
+        ssu        git.merproject.org:mer-core/ssu  master  -declarative
+        mpris-qt5  github.com:nemomobile/qtmpris    master  -qml-plugin
+
+    A remote configuration file may be specified with an HTTP(S)/FTP(S) URL in
+    which case curl(1) will be used for download. Note that curl does not
+    enable netrc(5) based authentication by default - add '--netrc' to your
+    ~/.curlrc if you wish to use netrc authentication.
+
+    Alternatively the configuration file can use the CSV format. See '--csv'.
+    This is useful e.g. for keeping the configuration in an Ethercalc
+    spreadsheet:
+
+        sdk-make-qmltypes --batch --csv
+            https://ethercalc.example.com/_/<SPREADSHEET-ID>/csv \
+            ~/projects
+
+
+OVERVIEW OF BATCH OPERATION
+    Each package is processed gradually, with these possible states:
+
+    TODO
+        Processing not started yet
+
+    CLONED
+        Reference repository was located under PROJECTS_DIR and it was cloned
+        under the working directory
+
+    MAKEFILES-READY
+        Depending on the project either qmake or CMake was run to produce
+        Makefiles
+
+    INSTALLED
+        Binary packages were installed to the emulator
+
+    DONE
+        The qmltypes files were updated by running \`make qmltypes\` in every
+        directory that contains at least one '*.qmltypes' file tracked by Git
+
+    FAIL
+        Package was marked as failed and will be excluded from processing
+
+LIMITATIONS
+    Only qmake- and CMake-based projects are supported.
+
+    Batch operation is only supported under Docker-based build engine running
+    on a Linux host.
+END
+}
+
+debug()
+{
+    :
+}
+
+info()
+{
+    echo "$*" >&2
+}
+
+warning()
+{
+    echo "Warning: $*" >&2
+}
+
+fatal()
+{
+    echo "Fatal: $*" >&2
+}
+
+configure()
```

Comment:
I'm not sure this is the best name for the function. It doesn't actually configure anything, it just checks some prerequisites.

Reply:
Checking for prerequisites is what configure scripts do first of all. Whether they can deal with particular prerequisite failing or not is just an additional level. Imagine it would install giturlparse instead of asking the user to do so and you wouldn't find it wrong.

Reply:
I didn't make the association with configure scripts. Sure, now that you put it like that, it makes sense.